sql/core/src/main/scala/org/apache/spark/sql/execution/command/DataWritingCommand.scala

```diff
@@ -57,4 +59,6 @@ trait DataWritingCommand extends RunnableCommand {
     val serializableHadoopConf = new SerializableConfiguration(hadoopConf)
     new BasicWriteJobStatsTracker(serializableHadoopConf, metrics)
   }
+
+  def run(sparkSession: SparkSession, children: Seq[SparkPlan]): Seq[Row]
```

Comment:
since we define children as `query :: Nil`, here we can just pass `query: SparkPlan`